docs/4.1. Behaviour - Clients.md

```diff
@@ -19,33 +19,93 @@ Clients SHOULD support a mechanism for providing the information required for OA
 containing the required details).
 
 Clients are RECOMMENDED to provide for the OAuth 2.0 Dynamic Client Registration Protocol [RFC 7591][RFC-7591] to allow
-for zero-configuration setup.
+for zero-configuration setup. NMOS Nodes SHOULD be capable of registering dynamically in order to limit the
+configuration associated with large numbers of Nodes coming online simultaneously.
+
+Clients that do not register with an explicit `token_endpoint_auth_method` parameter will be registered as a
+Confidential Client and issued with a client secret. Public clients MUST specify a value of "none" for this parameter
+to notify the Authorization Server that is is a public client.
 
 ## Client Credentials
 
 Clients operating with a client secret SHOULD support using HTTP Basic Authentication, as per Section 2 of
 [RFC 2617][RFC-2617], to authenticate with the Authorization Server in the manner described in Section 2.3.1 of
 [RFC 6749][RFC-6749].
 
-Client credentials should be stored by the client in a safe location in non-volatile memory in case of a device restart
-to prevent duplicate registrations.
+Client credentials SHOULD be stored by the client in a safe, permission-restricted, location in non-volatile memory in
+case of a device restart to prevent duplicate registrations. Client secrets SHOULD be hashed before being stored to
```

Comment:
Does the client secret not get passed in requests? If so perhaps 'encrypted' would be more appropriate? A hash sounds like a one-way procedure.